lib/Common/Memory/Recycler.cpp

```diff
@@ -947,12 +947,12 @@ Recycler::IsSweeping() const
 }
 
 void
-Recycler::SetIsScriptActive(bool isScriptActive)
+Recycler::SetIsScriptActive(bool isScriptActive, bool doCleanup)
 {
     Assert(this->isInScript);
     Assert(this->isScriptActive != isScriptActive);
     this->isScriptActive = isScriptActive;
-    if (isScriptActive)
+    if (isScriptActive && doCleanup)
```

Comment:
Sure. If we don't push the timer, it is more likely the `tick > tickCountNextDispose`